internal/cluster/workflow/health_check_activity_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/testsuite"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/testing_frameworks/integration"
+
+	"github.com/banzaicloud/pipeline/pkg/k8sclient"
+)
+
+// nolint: gochecknoglobals
+var healthCheckTestActivity = HealthCheckActivity{}
+
+func testHealthCheckActivityExecute(ctx context.Context, input HealthCheckActivityInput) error {
+	return healthCheckTestActivity.Execute(ctx, input)
+}
+
+// nolint: gochecknoinits
+func init() {
+	activity.RegisterWithOptions(testHealthCheckActivityExecute, activity.RegisterOptions{Name: HealthCheckActivityName})
+}
+
+type HealthCheckActivityTestSuite struct {
+	suite.Suite
+	testsuite.WorkflowTestSuite
+
+	env *testsuite.TestActivityEnvironment
+
+	controlPlane *integration.ControlPlane
+
+	client kubernetes.Interface
+}
+
+func testHealthCheckActivity(t *testing.T) {
+	if os.Getenv("TEST_ASSET_KUBE_APISERVER") == "" || os.Getenv("TEST_ASSET_ETCD") == "" {
+		t.Skip("control plane binaries are missing")
+	}
```

Comment:
**Question**: is this supposed to be an integration test only or should it also be a mocked unit test?
(In case of the former I would suggest having a mocked unit test as well besides the integration test.)

Are we testing against some expected environment here?
(If we do so I would suggest abstracting it away and mocking it, but not sure about the context/scope yet.)

Why is it required to initialize `s.client` and can it be mocked away?
(What calls it? The health checker?)

(This might be easier to discuss IRL.)

Reply:
In the codebase, I found some integration tests, and they were implemented in this way.

Reply:
All right, then I have no objections.